### PR TITLE
Channel monad (with little other improvements)

### PR DESCRIPTION
**Pull request by niwibe**
- Now with clojurescript tests.
- With monad laws tests.

### CODE REVIEW

**Comment by dialelo** [comment]
Looks good so far, looking forward to the CLJS tests and docs. I also think we should complete the docs before the next major release, not just for the chan monad but for the rest of them.

**Comment by niwibe** [comment]
I agree with you about docs, would be awesome have everything documented.

Now, this pr is ready and includes documentation for channel ;)

File: doc/cats.adoc [hunk 1]
```diff
@@ -680,32 +656,102 @@ However, if you have better examples to explain the state monad, documentation f
 any other contribution is always welcome.
 
 
-Reader
-~~~~~~
+=== Channel
+
+In asynchronous environments with clojure and clojurescript we tends to use core.async, because it
+a very powerfull abstraction.
```
Comment (dialelo) [inline doc/cats.adoc]: "it is a"

**Comment by dialelo** [comment]
I pushed a commit with some wording improvements to the docs, if you are happy with them feel free to merge. This is a great addition to the library, awesome work!

**Pull request merged**